sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -77,7 +77,8 @@ object Cast {
         resolvableNullability(fn || forceNullable(fromType, toType), tn)
 
     case (MapType(fromKey, fromValue, fn), MapType(toKey, toValue, tn)) =>
-      canCast(fromKey, toKey) && canCastMapKeyNullSafe(fromKey, toKey) &&
+      canCast(fromKey, toKey) &&
+        (!forceNullable(fromKey, toKey)) &&
```

Comment:
nit: can we remove the outermost parentheses?

Reply:
This one was a clean revert from this line (https://github.com/apache/spark/commit/d7b97a1d0daf65710317321490a833f696a46f21#diff-258b71121d8d168e4d53cb5b6dc53ffeL81). I would keep this line as is ..